frontend/src/components/form/tiptap/TiptapEditor.vue

```diff
@@ -5,82 +5,70 @@
-            @click="editor.chain().focus().toggleHeading({ level: 3 }).run()"
-          >
-            <v-icon>mdi-format-header-3</v-icon>
-          </v-btn>
-        </v-item-group>
-        <div class="mx-1" />
-        -->
-        <v-item-group class="v-btn-toggle v-btn-toggle--dense" multiple>
-          <v-btn
-            :class="editor.isActive('bold') ? 'v-item--active v-btn--active' : ''"
-            @click="editor.chain().focus().toggleBold().run()"
-          >
-            <v-icon dense>mdi-format-bold</v-icon>
-          </v-btn>
-          <v-btn
-            :class="editor.isActive('italic') ? 'v-item--active v-btn--active' : ''"
-            @click="editor.chain().focus().toggleItalic().run()"
-          >
-            <v-icon dense>mdi-format-italic</v-icon>
-          </v-btn>
-          <v-btn
-            :class="editor.isActive('underline') ? 'v-item--active v-btn--active' : ''"
-            @click="editor.chain().focus().toggleUnderline().run()"
-          >
-            <v-icon dense>mdi-format-underline</v-icon>
-          </v-btn>
-          <v-btn
-            :class="editor.isActive('strike') ? 'v-item--active v-btn--active' : ''"
-            @click="editor.chain().focus().toggleStrike().run()"
-          >
-            <v-icon dense>mdi-format-strikethrough</v-icon>
-          </v-btn>
-        </v-item-group>
+      <v-toolbar class="ec-tiptap-toolbar" dense>
+        <TiptapToolbarButton
+          icon="mdi-format-bold"
+          :class="editor.isActive('bold') ? 'v-item--active v-btn--active' : ''"
+          @click="editor.chain().focus().toggleBold().run()"
+        />
+        <TiptapToolbarButton
+          icon="mdi-format-italic"
+          :class="editor.isActive('italic') ? 'v-item--active v-btn--active' : ''"
+          @click="editor.chain().focus().toggleItalic().run()"
+        />
+        <TiptapToolbarButton
+          icon="mdi-format-underline"
+          :class="editor.isActive('underline') ? 'v-item--active v-btn--active' : ''"
+          @click="editor.chain().focus().toggleUnderline().run()"
+        />
+        <TiptapToolbarButton
+          icon="mdi-format-strikethrough"
+          :class="editor.isActive('strike') ? 'v-item--active v-btn--active' : ''"
+          @click="editor.chain().focus().toggleStrike().run()"
+        />
+
+        <v-divider vertical class="mx-1" />
+
+        <TiptapToolbarButton
+          icon="mdi-format-list-bulleted"
+          :class="editor.isActive('bulletList') ? 'v-item--active v-btn--active' : ''"
+          @click="editor.chain().focus().toggleBulletList().run()"
+        />
+        <TiptapToolbarButton
+          icon="mdi-format-list-numbered"
+          :class="editor.isActive('orderedList') ? 'v-item--active v-btn--active' : ''"
+          @click="editor.chain().focus().toggleOrderedList().run()"
+        />
+
+        <template
+          v-if="
+            editor.can().sinkListItem('listItem') || editor.can().liftListItem('listItem')
```

Comment:
i would toggle the sink and lift list item buttons separately (or disable them), because if you cannot sink/lift an item, nothing happens if you press the button.

Reply:
My implemented solution has the best UX, as these fields are only displayed if they are really necessary. Less is more.